src/carousel.css.ts

```diff
@@ -0,0 +1,334 @@
+        alignItems: 'center',
+        justifyContent: 'center',
+        borderRadius: '50%',
+        backgroundColor: skinVars.colors.backgroundContainer,
+        width: arrowButtonSize,
+        height: arrowButtonSize,
+    }),
+    {
+        transition: 'opacity 0.2s',
+        selectors: {
+            '&[disabled]': {
+                opacity: 0,
+            },
+        },
+        '@media': {
+            // don't show carrousel arrow buttons in touch devices, just regular horizontal scroll
+            [mq.touchableOnly]: {
+                display: 'none',
+            },
+        },
+    },
+]);
+
+export const carouselContainer = sprinkles({
+    // This minWidth value is a workaround to solve an issue when the page is rendered in a hidden webview
+    // in that case the window size is reported as 0 and the scroll snap is placed at the wrong slide
+    minWidth: 64,
+    position: 'relative',
+});
+
+export const carousel = style([
+    hideScrollbar,
+    sprinkles({
+        display: 'flex',
+        minWidth: '100%',
+    }),
+    {
+        overflowX: 'auto',
+    },
+]);
+
+export const carouselWithScroll = style({
+    margin: `0 calc(${responsiveLayoutVars.sideMargin} * -1)`,
+});
+
+export const centeredCarousel = style({
+    '@media': {
+        [mq.tabletOrSmaller]: {
+            ':before': {
+                content: '""',
+                flexShrink: 0,
+                display: 'block',
+                width: '25%',
+            },
+            ':after': {
+                content: '""',
+                flexShrink: 0,
+                display: 'block',
+                width: '25%',
+            },
+        },
+    },
+});
+
+const itemsPerPageMobile = createVar();
+const itemsPerPageDesktop = createVar();
+const mobilePageOffsetPrev = createVar();
+const mobilePageOffsetNext = createVar();
+const gap = createVar();
+
+export const vars = {
+    itemsPerPageMobile,
+    itemsPerPageDesktop,
+    mobilePageOffsetPrev,
+    mobilePageOffsetNext,
+    gap,
+};
+
+export const carouselItem = style({
+    scrollSnapAlign: 'start',
+    flexShrink: 0,
+    width: `calc(1 / ${itemsPerPageDesktop} * 100% + ${gap} / ${itemsPerPageDesktop} * 1px)`,
+    scrollMargin: `calc(-1px * ${gap})`,
+    ':first-child': {
+        width: `calc(1 / ${itemsPerPageDesktop} * 100% - (${gap} * (${itemsPerPageDesktop} - 1)) / ${itemsPerPageDesktop} * 1px)`,
+        scrollMargin: 0,
+    },
+    '@media': {
+        [mq.tabletOrSmaller]: {
+            width: `calc(1 / ${itemsPerPageMobile} * 100% + ${gap} / ${itemsPerPageMobile} * 1px)`,
+
+            ':first-child': {
+                width: `calc(1 / ${itemsPerPageMobile} * 100% - ${gap} * (${itemsPerPageMobile} - 1) / ${itemsPerPageMobile} * 1px)`,
+            },
+
+            scrollSnapAlign: 'start',
+            scrollMargin: `calc(${mobilePageOffsetPrev} * 1px)`,
+
+            selectors: {
+                [`${carouselWithScroll}:not(${centeredCarousel}) &`]: {
```

Comment:
I had some issues with this approach, if `centeredCarousel` in the future has also a sprinkles (just because it changes) this selector wont work, because will be something like this `:not(.style_blabla .sprinkles_xxx)` that does not work on CSS. 
I found safer create empty clases `const notSelector = style({})`to cover this usages, and avoid broke it without notice in the future...
WDYT?

Reply:
are you sure about that? I'd say it'd still work, but I'll re-check it

Reply:
yep, had this issue this Chip component, in this case this works because `centeredCarousel` only has a style (no sprinkles, no base class) so only one string class is generated `.centeredCarousel_xxx`, but if you add a base case a sprinkles, that this generate more than one class and this :not() will fail

Reply:
checked and it works without problems, VE does the magic and only include one class

Reply:
![image](https://user-images.githubusercontent.com/1849576/202684166-1a03f58f-bfb3-4725-91c3-0c5d0ea6e480.png)